src/calendar/index.js

```diff
@@ -123,47 +125,38 @@ class Calendar extends Component {
   }
 
   pressDay = date => {
-    this._handleDayInteraction(date, this.props.onDayPress);
+    this.handleDayInteraction(date, this.props.onDayPress);
   };
 
   longPressDay = date => {
-    this._handleDayInteraction(date, this.props.onDayLongPress);
+    this.handleDayInteraction(date, this.props.onDayLongPress);
   };
 
-  getDateMarking(day) {
-    const {markedDates} = this.props;
-
-    if (!markedDates) {
-      return false;
-    }
-
-    const dates = markedDates[day.toString('yyyy-MM-dd')] || EmptyArray;
-
-    if (dates.length || dates) {
-      return dates;
-    } else {
-      return false;
-    }
-  }
-
+  //TODO: this method has a duplication in Week component - consider moving to presenter/state manager
   getState(day) {
-    const {disabledByDefault} = this.props;
+    const {disabledByDefault, context} = this.props;
     const minDate = parseDate(this.props.minDate);
     const maxDate = parseDate(this.props.maxDate);
     let state = '';
-
+    
+    if (context?.date === toMarkingFormat(day)) {
+      state = 'selected';
+    } else if (dateutils.isToday(day)) {
+      state = 'today';
+    }
```

Comment:
Same comment I wrote in Week component. 
And maybe we should have a method for `getState`, the code is the same

Reply:
I know it's the same, I wrote a TODO to extract it to a state manager file